core/src/env_platform/env_service.bal

```diff
@@ -82,6 +99,45 @@ service envservice on ep0 {
     }
     resource function putApplicationById(http:Caller caller, http:Request req, string applicationId, TreeRemovalForm body) returns error? {
 
+        http:Response response = new;
+        string authHeader = req.getHeader("Authorization");
+        [string, string]|error userInfoFromJWT = getUserInfoFromJWT(authHeader);
+        if (userInfoFromJWT is [string, string]) {
+            [string, string] [userId, userType] = userInfoFromJWT;
+            boolean|error applicationBelongsToUserResult = applicationBelongsToUser(applicationId, userId);
+            if (applicationBelongsToUserResult is error) {
+                response.statusCode = http:STATUS_NOT_FOUND;
+                if (applicationBelongsToUserResult.reason() == "No applications") {
+                    response.setPayload({"reason": "No applications for given user."});
+                } else {
+                    response.setPayload({"reason": "No such user."});
+                }
+            } else {
+                // If application is found.
+                if (applicationBelongsToUserResult) {
+                    boolean|error application = updateApplication(body, applicationId);
+                    if (application is error) {
+                        response.statusCode = http:STATUS_INTERNAL_SERVER_ERROR;
+                        response.setPayload({"reason": "Application is not updated."});
+                    } else {
+                        // If application is updated.
+                        if (application) {
+                            response.statusCode = http:STATUS_OK;
+                            response.setPayload({"reason": "Application is updated."});
+                        } else {
+                            response.statusCode = http:STATUS_INTERNAL_SERVER_ERROR;
+                            response.setPayload({"reason": "Application is not updated."});
```

Comment:
Why do we need another 500 block after checking whether application is returning an error in line number 119? If we just need to check for a null value, might as well have it in the same **if** condition.

Reply:
To check whether the application is updated or not. Or it can be done inside the first if as you mentioned. What would be the better option? @deshankoswatte @sanjulamadurapperuma

Reply:
I think it is better to do the validation present in the else block in the first if condition itself, since the error message returned is the same.